### PR TITLE
Add jarm docs

### PR DESCRIPTION
**Pull request by divyaamunugama**
Related issue: https://github.com/wso2/product-is/issues/15943

### CODE REVIEW

File: en/docs/references/concepts/authorization/jarm.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
+# JWT Secured Authorization Response Mode (JARM) for OAuth 2.0
+
+The [JWT Secured Authorization Response Mode for OAuth 2.0 (JARM) specification](https://openid.net/specs/openid-financial-api-jarm-ID1.html) defines new JWT-based modes to encode OAuth2 authorization responses. These modes allow clients to request authorization response parameters and additional data in JWT format.
+
+## Authorization flow
```
Comment (divyaamunugama) [inline en/docs/references/concepts/authorization/jarm.md:5]: Shall we update this as `JARM authorization flow`

File: en/docs/references/concepts/authorization/jarm.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+                <li><code>form_post</code></li>
+            </ul>
+            In the default authorization flow (for default response modes), the authorization response parameters, such as the authorization code and ID token, are sent as human-readable plain text to the redirect URL, as shown below.</br></br>
+            <code>https://&lt;CLIENT_HOST&gt;/redirects/redirect1?code=&lt;AUTH_CODE&gt;&session_state=&lt;SESSION_STATE&gt;</code></br></br>
+            The JARM specification is used to send this response more securely by encoding the response parameters in a JWT.
+        </td>
+    </tr>
+</table>
+
+## Enable JARM
```
Comment (divyaamunugama) [inline en/docs/references/concepts/authorization/jarm.md:58]: Took this to the top before `Using JARM`. hope it's okay?

File: en/docs/references/concepts/authorization/jarm.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+## Using JARM
+
+When [JARM is enabled](#enable-jarm), clients can request authorization response parameters along with additional data in JWT format instead of plain text.
+
+The specification defines the following response modes, which you can specify using the `response_mode` parameter in the authorization request.
+
+- [response_mode=query.jwt](#queryjwt)
+- [response_mode=fragment.jwt](#fragmentjwt)
+- [response_mode=form_post.jwt](#form_postjwt)
+- [response_mode=jwt](#jwt)
```
Comment (divyaamunugama) [inline en/docs/references/concepts/authorization/jarm.md:94]: Added this as a new section, is it correct to add the response format like this?

File: en/docs/references/concepts/authorization/jarm.md (hunk 1 above)
Comment (chamathns) [reply en/docs/references/concepts/authorization/jarm.md:5]: +1

File: en/docs/references/concepts/authorization/jarm.md (hunk 2 above)
Comment (chamathns) [reply en/docs/references/concepts/authorization/jarm.md:58]: yep. that's actually better

File: en/docs/references/concepts/authorization/jarm.md (hunk 3 above)
Comment (chamathns) [reply en/docs/references/concepts/authorization/jarm.md:94]: yeah +1

**Review by chamathns — APPROVED** [review]

File: en/docs/references/concepts/authorization/jarm.md (hunk 1 above)
Comment (divyaamunugama) [inline en/docs/references/concepts/authorization/jarm.md:5]:
```suggestion
## JARM authorization flow
```

**Review by himeshsiriwardana — APPROVED** [review]

**Pull request merged**